org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/points/ValidateResultsView.java

```diff
@@ -0,0 +1,239 @@
+	 * @param parent
+	 */
+	@Override
+	public void createPartControl(Composite parent) {
+		
+		Composite container = new Composite(parent, SWT.NONE);
+		container.setLayout(new GridLayout(1, false));
+		
+		this.text = new StyledText(container, SWT.MULTI | SWT.H_SCROLL | SWT.V_SCROLL);
+		text.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		text.setBackground(text.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+		text.getParent().layout(new Control[]{text});
+		
+		// We force the scan view to exist. It might be the one to return the compound model
+		// that we will use.
+		ScanningPerspective.createKeyPlayers();
+		
+        createActions();
+	}
+	
+	/**
+	 * Create the actions
+	 */
+	private void createActions() {
+		final IContributionManager man = getViewSite().getActionBars().getToolBarManager();
+
+		final Action restart = new Action("Clear", Activator.getImageDescriptor("icons/layers-stack.png")) {
+			public void run() {
+				clear();
+			}
+		};
+		man.add(restart);
+
+		final MenuManager menuMan = new MenuManager();
+		menuMan.add(restart);
+		
+		text.setMenu(menuMan.createContextMenu(text));
+	}
+
+	@Override
+	public void dispose() {
+		if (PageUtil.getPage(getSite())!=null) PageUtil.getPage(getSite()).removeSelectionListener(this);
+		super.dispose();
+	}
+	
+	private void clear() {
+		setThreadSafeText(text, new StyledString(""));
+	}
+	
+	@Override
+	public void selectionChanged(IWorkbenchPart part, ISelection selection) {
+		if (selection instanceof IStructuredSelection) {
+			Object ob = ((IStructuredSelection)selection).getFirstElement();
+
+			if (ob instanceof ValidateResults) {
+				ValidateResults validationResults  = (ValidateResults)ob;
+				update(validationResults);
+			}
+
+		}
+	}
+
+	/**
+	 * Method for updating the display with the validation results
+	 * @param results the ValidationResults object
+	 */
+	public void update(ValidateResults results) {
+		if (results == null) return;
+		
+		try {
+
+    		StyledString styledString = new StyledString();
+        	
+			styledString.append("Results received from '");
+			styledString.append(results.getDeviceName(), FontStyler.BOLD);
+			styledString.append("' at ");
+			styledString.append(new Date(System.currentTimeMillis()).toString(), StyledString.COUNTER_STYLER);
+			styledString.append("\n");
+        	
+			styledString.append(appendResultsToStyledString(results));
+
+            setThreadSafeText(text, styledString);
+            return; 
+
+		} catch (Exception ne) {
+			logger.error("Cannot create validation results", ne);
+			if (ne.getMessage()!=null) {
+				setThreadSafeText(text, ne.getMessage());
+			} else {
+				setThreadSafeText(text, ne.toString());
+			}
+		}
+	}
+	
+	/**
+	 * Turn the results into a styled string for display to user, highlighting key values
+	 * @param results The ValidationResults
+	 * @return a StyledString to display on the view
+	 */
+	private StyledString appendResultsToStyledString(ValidateResults results) {

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3776)